apps/admin-gui/src/app/shared/components/dialogs/copy-sponsored-members-dialog/copy-sponsored-members-dialog.component.ts

```diff
@@ -0,0 +1,146 @@
+  styleUrls: ['./copy-sponsored-members-dialog.component.scss'],
+})
+export class CopySponsoredMembersDialogComponent implements OnInit {
+  loading = false;
+  theme: string;
+  sponsorType: string;
+  tableId = TABLE_ADD_SPONSORED_MEMBERS;
+  voSponsorsTarget: RichUser[];
+  voSponsorsSource: RichUser[];
+  sourceSponsor: User;
+  targetSponsor: User;
+  sponsoredMembers: MemberWithSponsors[];
+  filteredSponsoredMembers: MemberWithSponsors[];
+  selection: SelectionModel<MemberWithSponsors> = new SelectionModel<MemberWithSponsors>(true, []);
+  sourceSponsorSelected = false;
+  filter: string;
+  expiration = 'never';
+  pickExpiration = false;
+  disableSelf = false;
+  isPerunAdmin = false;
+  private attrNames: string[] = [Urns.USER_DEF_PREFERRED_MAIL];
+  constructor(
+    private dialogRef: MatDialogRef<CopySponsoredMembersDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: CopySponsoredMembersDialogData,
+    private store: StoreService,
+    private membersService: MembersManagerService,
+    private notificator: NotificatorService,
+    private translate: PerunTranslateService,
+    private auth: GuiAuthResolver
+  ) {}
+
+  ngOnInit(): void {
+    this.attrNames = this.attrNames.concat(this.store.getLoginAttributeNames());
+    this.membersService
+      .getAllSponsoredMembersAndTheirSponsors(this.data.voId, this.attrNames)
+      .subscribe({
+        next: (members) => {
+          this.sponsoredMembers = members;
+        },
+      });
+    this.voSponsorsTarget = this.data.voSponsors;
+    this.voSponsorsSource = this.data.voSponsors;
+    this.isPerunAdmin = this.auth.isPerunAdmin();
+    // if not PERUNADMIN user cannot copy to anyone else than self, so do not allow selecting self as source sponsor
+    if (!this.isPerunAdmin) {
+      this.voSponsorsSource = this.voSponsorsSource.filter(
+        (voSponsor) => voSponsor.id !== this.store.getPerunPrincipal().user.id
+      );
+    }
+    this.theme = this.data.theme;
+  }
+
+  sourceSponsorChanged(user: User): void {
+    this.loading = true;
+    this.sourceSponsor = user;
+    this.voSponsorsTarget = this.data.voSponsors.filter(
+      (sponsor) => sponsor.id !== this.sourceSponsor.id
+    );
+    // disable copying to self only if PERUNADMIN (it is not possible to choose self as source sponsor without PERUNADMIN anyway)
+    if (this.isPerunAdmin) {
+      this.disableSelf = this.store.getPerunPrincipal().user.id === this.sourceSponsor.id;
+    }
+    this.sourceSponsorSelected = true;
+    this.selection.clear();
+    this.filteredSponsoredMembers = this.sponsoredMembers.filter((member) =>
+      member.sponsors.map((sponsor) => sponsor.user.id).includes(this.sourceSponsor.id)
+    );
+    this.loading = false;
+  }
+
+  sponsor(members: MemberWithSponsors[]): void {
+    this.loading = true;
+
+    const sponsor =
+      this.sponsorType === 'self' ? this.store.getPerunPrincipal().user : this.targetSponsor;
```

Comment:
There is a problem with role `SPONSOR` when I want to copy sponsored members.
I think that the problem is that for role `SPONSOR` there is no default value for `sponsorType`, so this condition is evaluated incorrectly. 
![image](https://user-images.githubusercontent.com/70371638/227145886-b3556ebb-e221-44b9-9033-ad73ff89c95c.png)

Reply:
fixed